beacon-chain/core/altair/sync_committee.go

```diff
@@ -115,3 +118,119 @@ func SyncCommitteeIndices(state iface.BeaconStateAltair, epoch types.Epoch) ([]t
 	}
 	return cIndices, nil
 }
+
+// AssignedToSyncCommittee returns true if input validator `i` is assigned to a sync committee at epoch `e`.
+//
+// Spec code:
+// def is_assigned_to_sync_committee(state: BeaconState,
+//                                  epoch: Epoch,
+//                                  validator_index: ValidatorIndex) -> bool:
+//    sync_committee_period = compute_sync_committee_period(epoch)
+//    current_epoch = get_current_epoch(state)
+//    current_sync_committee_period = compute_sync_committee_period(current_epoch)
+//    next_sync_committee_period = current_sync_committee_period + 1
+//    assert sync_committee_period in (current_sync_committee_period, next_sync_committee_period)
+//
+//    pubkey = state.validators[validator_index].pubkey
+//    if sync_committee_period == current_sync_committee_period:
+//        return pubkey in state.current_sync_committee.pubkeys
+//    else:  # sync_committee_period == next_sync_committee_period
+//        return pubkey in state.next_sync_committee.pubkeys
+func AssignedToSyncCommittee(
+	state iface.BeaconStateAltair,
+	epoch types.Epoch,
+	i types.ValidatorIndex,
+) (bool, error) {
+	p := SyncCommitteePeriod(epoch)
+	currentEpoch := helpers.CurrentEpoch(state)
+	currentPeriod := SyncCommitteePeriod(currentEpoch)
+	nextEpoch := currentPeriod + 1
+
+	if p != currentPeriod && p != nextEpoch {
+		return false, errors.New("epoch period is not current period or next period in state")
```

Comment:
Info here about the values would be useful in the error

Reply:
Addressed. Thanks!